### PR TITLE
QWfilter

### PR DESCRIPTION
**Pull request by andresmmera**
Added quarter wavelength bandpass and notch filters to qucs-filter tool. Both the transmission line and the microstrip are available.

* I have made a pull request before, but something seemed to be wrong so I had to pull again. Excuse me, I am a newbie on Github

### CODE REVIEW

**Comment by guitorri** [comment]
Thank you for this.
I will close the other pull request.
Please also add the new sources to the Autotools build system.
https://github.com/Qucs/qucs/blob/master/qucs/qucs-filter/Makefile.am#L32
https://github.com/Qucs/qucs/blob/master/qucs/qucs-filter/Makefile.am#L57

**Comment by ra3xdh** [comment]
@andresmmera I have tested it. All works fine. I think it could be merged. Could you also add a small documentation in `qucs-doc` with mathematical background description of your calculation methods?

**Comment by guitorri** [comment]
Please squash the commits to a single commit and add a descriptive [commit message](https://github.com/Qucs/qucs/wiki/Contribution#rules-about-commits).
This looks like a backport from QucsStudio...

**Comment by andresmmera** [comment]
@guitorri I've taken the old commit messages and I put them in the new commit. According to https://github.com/Islandora/islandora/wiki/Git-Guidelines-and-Best-Practices, my commit message should have multi-line appearance, but it doesn't.

@ra3xdh I have already added the design equations to '/qucs-doc/technical/synthesis.tex'. However, it seems that most of the filter topologies have not been included in 'technical.pdf'

**Comment by andresmmera** [comment]
Both Travis CI and AppVeyor have failed. I have only modified 'synthesis.tex' and I have added an .eps figure... Did I do something wrong?

**Comment by guitorri** [comment]
It seems that SoureForge is down again.... I is failing to download necessary stuff. I will try to implement a fallback.

**Comment by guitorri** [comment]
Your commit message seems odd. I thing the `\` escaping allow you to type a multiline message on the terminal, but it does not translate to an newline. Have you tried setting up a text editor in Git? Can you try to `amend` it please? It is also fine if you squash everything to a single commit.

If you want to fix the missing docs, please open a new pull request.

**Comment by in3otd** [comment]
regarding the doc, while you are there you can fix [this typo] (https://github.com/andresmmera/qucs/commit/a6a8956681b8b2569d39e283ab326655563c6d6f#diff-2cc0596a28ecb7e0609a53e35225b56dR223) ("approxiation").
In the text "i-esim" should probably be "i-th" (usually written with the 'i' in math mode, `$i$-th") but I'm not a native English speaker.

Note that we too have some commit guidelines [here] (https://github.com/Qucs/qucs/wiki/Contribution#rules-about-commits) :grin: (very similar to the ones you linked, of course)

**Comment by andresmmera** [comment]
@guitorri  I think that now the commit message is ok. It is far more useful to use nano than the -m option...
@in3otd Misspelling stuff corrected :-)

**Comment by guitorri** [comment]
Almost there :smile:. You are adding a long message on the short summary

You need a message like
```
Short summary (no longer than 50 chars)
       [blank line]
The message.... that you have...
The message.... that you have...
```

Can you please squash the 6 commits into 1? 

Something like this will open your editor with your last 6 commits:
`git rebase -i QWFilter~6`

Change the work `pick` to `squash` (just `s` will do) for all the commits below `5f05b6e` (your first commit). Save and close. Another editor will pop up so that you can edit the final commit message as suggested above. 

You will need to push with force (add option -f) update your branch.

**Comment by andresmmera** [comment]
@guitorri Well,... the first time I read 'squash' I didn't know exactly what I was supposed to do (as I said I'm new on Github). Anyway, I think that now it is ok, isn't it? Many thanks for your patience :-)
P.S.: AppVeyor keeps failing

**Comment by in3otd** [comment]
@andresmmera don't worry, we have been all there :grin: .

If you have time, please give some feedback on [the guidelines] (https://github.com/Qucs/qucs/wiki/Contribution) (the `squash` part is still missing there, actually), it is intended also to help new contributors to ramp up - let us know if something is not clear or missing, so we can improve it.

**Comment by andresmmera** [comment]
Ok, here is some feedback:

The development guideline is quite complete and easy to follow. However, I think it would be even better if you include:

1) A simple diagram of the Github workflow. I think it can help to visualize how Github works. Maybe, something like this [1].
2) A Github cheatsheet. https://goo.gl/EnGDt7
3) It would be perfect if you could describe in more detail how to handle several branches. I mean, it is quite common to work in different branches at the same time, so maybe you could include information regarding how to list branches (git branch), how to change the working branch (git checkout <branch_name>), etc..

[1] https://dl.dropboxusercontent.com/u/21294824/GithubWorkflow.png
[2] https://dl.dropboxusercontent.com/u/21294824/Github%20flow.odg    (Source file)

**Comment by guitorri** [comment]
The commit message is great. It is very detailed :smile: . Thanks.
I don't know what is going on with Appveryor. Last time it was stuck with downloads from SourceForge (which seems to be down again). I will take a took at it.

**Comment by guitorri** [comment]
@andresmmera can you please rebase/resolve the conflict and push again? This will also re-run the CI machines...

**Comment by andresmmera** [comment]
@guitorri Just checking, should I do the following?
git checkout QWFilter
git rebase QWFilter
git push origin QWFilter -f

**Comment by guitorri** [comment]
Update master first, and rebase on master. Something like this
```
git checkout master
git pull upstream master
git checkout QWFilter
git rebase master
[fix conflicts, add, commit ... do what is needed :) ]
git push -f origin QWFilter
```
Where `upstream` is the Qucs repository/remote and `origin` is your fork/remote.

You might need to add Qucs as a remote first. Something like:
`git remote add upstream https://github.com/Qucs/qucs.git`

**Comment by andresmmera** [comment]
@guitorri Ok, I think now it's fixed. There was a conflict on 'technical/literature.bib'.
Let's see if it passes the tests...

Thanks for you help :-)

File: qucs-core/build-aux/ar-lib [hunk 1]
```diff
@@ -0,0 +1,270 @@
+#! /bin/sh
```
Comment (guitorri) [inline qucs-core/build-aux/ar-lib]: I guess this file should not be in the repository.

**Comment by andresmmera** [comment]
Ok, I've just removed it.

**Comment by andresmmera** [comment]
Hello all, 
After squashing the old (and compilable) commits, AppVeyor gave this error:

*Build started
git clone -q https://github.com/Qucs/qucs.git C:\projects\qucs
fatal: early EOF
fatal: The remote end hung up unexpectedly
fatal: index-pack failed
error: RPC failed; result=18, HTTP code = 200
Command exited with code 128*

I guess this kind of errors do not depend on the code itself but on the AppVeyor engine, am I right?

**Comment by guitorri** [comment]
Hum... look like a glitch either in GitHub or AppVeyor. I restarted the build...

**Comment by guitorri** [comment]
Merging this.
Thank you @andresmmera, and feel free to edit the wiki w.r.t Git usage :wink:

**Comment by andresmmera** [comment]
Oh. Ok, thanks :-)

**Pull request merged**